src/utils/messenger.h

```diff
@@ -161,7 +161,7 @@ class Messenger {
         TimeLine t("Messenger.reduceAdd");
 
 #ifdef USE_SHM
-        if (sizeof(T) * count > pshm->getSHMSize() || !localRanksFlag) {
+        if (!localRanksFlag || (pshm != nullptr && sizeof(T) * count > pshm->getSHMSize())) {
```

Comment:
Do we also need to add a condition "pshm == nullptr" ?

Reply:
It’s a good suggestion. Maybe it can be used to check whether shared memory is allocated successfully. I’ll think about it later.